session_security/middleware.py

```diff
@@ -62,7 +64,11 @@ def process_request(self, request):
         delta = now - get_last_activity(request.session)
         expire_seconds = self.get_expire_seconds(request)
         if delta >= timedelta(seconds=expire_seconds):
+            request._messages._queued_messages = []
+            messages.info(request, 'Your session has expired (%s). Please login again.' % delta)
             logout(request)
+            if request.is_ajax():
+                return defaults.http.HttpResponseForbidden('403 Forbidden: %s' % msg, content_type='text/html')
```

Comment:
what's that supposed to do

Reply:
If session is expired, a ajax call will get the righ feedback information. That's response.responseText will be '403 Forbidden: %s' % msg

Reply:
makes sense